# Motivacao

Estava assistindo a entrevista do Mario Filho com Leticia, e Ela mencionou esses 100 Exercicios do canal "Curso em Video".
<br> Hoje (03/10) da tarde, decidi me desafiar com esses exercicios, e quero ver em quantos dias vou conseguir resolver todos.
<br> Vou colocar a descricao do exercicio tambem, a solucao sera totalmente minha propria.
<br> **Link: https://www.youtube.com/watch?v=nIHq1MtJaKs&list=PLHz_AreHm4dm6wYOIW20Nyg12TAjmMGT-**

## Questao 1:

### Cria um programa que escreva "Ola, Mundo!" na tela

In [1]:
print("Ola, Mundo!")

Ola, Mundo!


## Questao 2 

### Faca um programa que leia o nome de uma pessoa e mostra uma mensagem de boa-vindas 

In [6]:
name = str(input("Favor escrever seu nome:"))
print("Bem vindo prezado %s!" %name)

Favor escrever seu nome:Mateus
Bem vindo prezado Mateus!


# Questao 3

## Crie um programa que leia dois numeros e mostra a soma entre eles

In [7]:
a = int(input("Digite o primeiro numero:"))
b = int(input("Digite o segundo numero:"))

print('A soma deles eh igual a : ', a+b)

Digite o primeiro numero:2
Digite o segundo numero:3
A soma deles eh igual a :  5


## Questao 4
 
### Faca um programa que leia algo pelo teclado  e mostra na tela seu tipo primitivo e todas as informacoes possiveis sobre ele

In [36]:
obj = input("Digite algum objeto, 0 para sair: ")

while obj != '0':
    try: 
        float(obj)
        if '.' in obj:
            print("Tipo primitivo do objeto eh: float")
        else:
            print("Tipo primitivo do objeto eh: int")
    except:
        try: 
            str(obj)
            if len(obj) == 1:
                print("Tipo primitivo do objeto eh: char")
            else:
                print("Tipo primitivo do objeto eh: str")
        except: 
            print("Nao existe mais outro tipo primitivo ?!")

    
    
    obj = input("Digite algum objeto, 0 para sair: ")

Digite algum objeto, 0 para sair: 23
Tipo primitivo do objeto eh: int
Digite algum objeto, 0 para sair: 2
Tipo primitivo do objeto eh: int
Digite algum objeto, 0 para sair: 2.0
Tipo primitivo do objeto eh: float
Digite algum objeto, 0 para sair: f
Tipo primitivo do objeto eh: char
Digite algum objeto, 0 para sair: fe
Tipo primitivo do objeto eh: str
Digite algum objeto, 0 para sair: "
Tipo primitivo do objeto eh: char
Digite algum objeto, 0 para sair: 0


## Questao 5

### Faca um programa que leia um numero inteiro e mostra na tela seu sucessor e seu antecessor

In [37]:
n = int(input("Digite o numero: "))

print("Seu antecessor eh %d e seu sucessor eh %d" %(n-1, n+1))

Digite o numero: 5
Seu antecessor eh 4 e seu sucessor eh 6


## Questao 6

### Crie um algoritmo que leia um numero e mostra seu dobro, triplo e raiz quadrada

In [39]:
num = float(input("Digite o numero: "))

dobro = 2*num
triplo = 3*num
raiz = num ** (1/2)

print(" Seu dobro eh %f,  \n Seu triplo eh %f, \n E, sua raiz eh %f" %(dobro, triplo, raiz))

Digite o numero: 5
Seu dobro eh 10.000000,  
 Seu triplo eh 15.000000, 
 E, sua raiz eh 2.236068


## Questao 7

### Desenvolva um programa que leia as duas notas de um aluno, calcula e mostra sua media

In [40]:
nota1 = int(input("Digite primeira nota do aluno: "))
nota2 = int(input("Digite segunda nota do aluno: "))

media = (nota1 + nota2)/2

print("A media das notas eh: ", media)

Digite primeira nota do aluno: 4
Digite segunda nota do aluno: 45
A media das notas eh:  24.5


## Questao 8

### Escreva um programa que leia um valor em metros e o exiba convertido em centimetros e milimetros

In [41]:
metros = float(input("Digite a distancia em metros: "))

cm = metros*100
mm = metros*1000

print("A distancia em cm eh %f e em mm eh %f" %(cm,mm))

Digite a distancia em metros: 45.6
A distancia em cm eh 4560.000000 e em mm eh 45600.000000


## Questao 9

### Faca um programa que leia um numero Inteiro qualquer e mostre na tela a sua tabuada

In [44]:
n = int(input('Digite o numero que ver sua tabuada: '))

print('=================================')
print("*********Tabuada de %d*********"%n)
print('=================================')

for i in range(10):
    print('          %d x %d = %d          '%(n,i+1, n*(i+1)))

Digite o numero que ver sua tabuada: 7
*********Tabuada de 7*********
          7 x 1 = 7          
          7 x 2 = 14          
          7 x 3 = 21          
          7 x 4 = 28          
          7 x 5 = 35          
          7 x 6 = 42          
          7 x 7 = 49          
          7 x 8 = 56          
          7 x 9 = 63          
          7 x 10 = 70          


## Questao 10

### Crie um programa que leia quanto dinheiro uma pessoa tem na carteira e mostre quantos dolares ele pode comprar. Considere US1.00 = RS3.27

In [46]:
rs = float(input("Dinnheiro em reais: "))
s = rs/3.27

print("Voce consegue comprar US%f dolares" %s)

Dinnheiro em reais: 56
Voce consegue comprar US17.125382 dolares


## Questao 11

### Faca um programa que leiaa largura e altura de uma parede em metros, calcula a sua area e quantidade de tinta necessaria para pinta-la, sabendo que cada litro de tinta pinta uma area de 2m^2. 

In [47]:
lar = float(input("Qual eh a largura (em metros) da parede? - "))
alt = float(input("Qual eh a altura (em metros) da parede? - "))

area = lar*alt
litros = area/2
print("Voce vai precisar de %f litros de tinta"%litros)

Qual eh a largura (em metros) da parede? - 3.5
Qual eh a altura (em metros) da parede? - 5
Voce vai precisar de 8.750000 litros de tinta


## Questao 12

### Faca um algoritmo que leia o preco de um produto e mostre seu novo preco, com 5% de desconto

In [49]:
preco = float(input('Preco sem desconto? - '))

n_preco = preco*0.95
print('Preco com 5%% de desconta sera: %f'%n_preco)

Preco sem desconto? - 45.67
Preco com 5% de desconta sera: 43.386500


## Questao 13

### Faca um algoritmo que leia o salario de um funcionario  e mostre seu novo salario, com 15% de aumento

In [50]:
sal = float(input('Salario sem aumento? - '))

n_salario = sal*1.15
print('Salario do funcionario com aumento de 15%% sera: %f'%n_salario)

Salario sem aumento? - 450
Salario do funcionario com aumento de 15% sera: 517.500000


## Questao 14

### Escreva um programa que converta uma temperatura digitada em Celsius e converta para Fahrenheit. 
### <center> F = C x 1.8 + 32 </center>

In [51]:
c = float(input('Graus em Celsius? - '))
f = c*1.8 + 32

print('Grau em fahrenheit sera : %f'%f)

Graus em Celsius? - 64
Grau em fahrenheit sera : 147.200000


## Questao 15

### Escreva um programa que pergunte a quantidade de KM percorrido por um carro alugado e quantidade de dias pelos quais ele foi alugado. Calcule o preco a pagar, sabendo que o carro custa RS60,00 por dia e RS0,15 por KM rodado.

In [52]:
km = float(input("Quantidade de KM percorrido? - "))
dias = int(input("Quantidade de dias usados? - "))

preco = 60*dias + km*0.15

print("O preco sera paga : %f"%preco)

Quantidade de KM percorrido? - 654
Quantidade de dias usados? - 34
O preco sera paga : 2138.100000


## Questao 16

###  Crie um programa que leia um numero Real qualquer pelo teclado e mostre na tela a sua porcao inteira.

 Ex: Digite um numero: 6.127
<br> O numero 6.127tem parte inteira 6.

In [53]:
numero = float(input("Digite um numero : "))

inteiro = int(numero)
print("O numero %f tem parte inteira %d."%(numero, inteiro))

Digite um numero : 5.24
O numero 5.240000 tem parte inteira 5.


## Questao 17

### Faca um programa que leia o comprimento do cateto oposto e do cateto adjacente de um triangulo retangulo, calcule e mostre o comprimento da hipotenusa.

In [54]:
a = float(input('Compritmento do cateto oposto? - '))
b = float(input('Comprimento do cateto adjacente? - '))

c = (a*a + b*b)**(1/2)

print("Sua hipotenusa sera : %f."%c)

Compritmento do cateto oposto? - 3
Comprimento do cateto adjacente? - 4
Sua hipotenusa sera : 5.000000.


## Questao 18

### Faca um programa que leia um angulo qualquer e mostre na tela o valor do seno, cosseno e tangente desse angulo.\

In [55]:
import math

angulo = float(input("Digite valor do angulo: "))

seno = math.sin(angulo)
cos = math.cos(angulo)
tan = math.tan(angulo)

print("Os valores do Sen, cosseno e tangente desse angulo: ")
print("Seno(%f) = %f"%(angulo,seno))
print("Cosseno(%f) = %f"%(angulo,cos))
print("Tanfente(%f) = %f"%(angulo,tan))

Digite valor do angulo: 45
Os valores do Sen, cosseno e tangente desse angulo: 
Seno(45.000000) = 0.850904
Cosseno(45.000000) = 0.525322
Tanfente(45.000000) = 1.619775


## Questao 19

### Um professor quer sortear um dos seus quatro alunos para apagaro quadro. Faca um programa que ajuda ele, lendo o nome deles e escrevendo o nome do escolhdo.

In [61]:
import random
aluno = [0]*4
for i in range(len(aluno)):
    nome = 'Qual eh o nome do %d esimo aluno? - '%(i+1)
    aluno[i] = input(nome)
    
escolhido = random.randint(0,3)
print("Nosso aluno escolhido eh %s"%aluno[escolhido])

Qual eh o nome do 1 esimo aluno? - M
Qual eh o nome do 2 esimo aluno? - A
Qual eh o nome do 3 esimo aluno? - T
Qual eh o nome do 4 esimo aluno? - E
Nosso aluno escolhido eh M
['E', 'M', 'T', 'A']


## Questao 20

### O mesmo professor do desafio anterior quer sortear  a ordem de apresentacao de trabalhos dos alunos. Faca um programa que leia o nome dos quatro alunos e mostre  a ordem sorteada.

In [62]:
random.shuffle(aluno)
for i in range(len(aluno)):
    print(" %d - esimo aluno para fazer sua apresentacao eh %s"%(i+1,aluno[i]))

 1 - esimo aluno para fazer sua apresentacao eh A
 2 - esimo aluno para fazer sua apresentacao eh T
 3 - esimo aluno para fazer sua apresentacao eh E
 4 - esimo aluno para fazer sua apresentacao eh M


## Questao 21

### Faca um programa em pyhton que abra e reproduza o audio de um arquivo mp3.

In [4]:
import IPython

musica = 'FurElise.mp3'

IPython.display.Audio(musica)

## Questao 22

### Crie um programa que leia o nome completo de uma pessoa e mostre:
<b> * O nome com todas as letras maiusculas e minusculas.
<br>    * Quantas letras ao todo (sem considerar espacos).
<br>    * Quantas letras tem o primeiro nome.

In [6]:
nome = input("Digite o nome: ")
print("Nome com letras minusculas: ", nome.lower())
print("Nome com letras maiusculas: ", nome.upper())
print("Nome tem %d letras."%(len(nome)+1))

Digite o nome: Mateus
Nome com letras minusculas:  mateus
Nome com letras maiusculas:  MATEUS
Nome tem 7 letras.


## Questao 23

### Faca um programa que leia um numero de 0 a 9999 e mostre tela cada um dos digitos separados.

**Ex:**
<br> Digite um numero: 1834
<br> Unidade: 4, dezena: 3, centena: 8, milhar: 1.

In [9]:
num = int(input("Digite um numero: "))
unidade = num%10
num = num//10
dezena = num%10
num = num//10
centena = num %10
num = num//10
milhar = num%10

print("Unidade: %d, dezena: %d, centena: %d, milhar: %d"%(unidade, dezena, centena, milhar))

Digite um numero: 1234
Unidade: 4, dezena: 3, centena: 2, milhar: 1


## Questao 24

### Crie um programa que leia o nome de uma cidade e diga se ela comeca ou nao com nome "SANTO". 

In [16]:
cidade = input("Escreva o nome da cidade:")
cidade = cidade.upper()
santo = cidade.split()
if santo[0] == "SANTO":
    print('Nome da cidade comece com SANTO')
else:
    print("Nome da cidade NAO comece com SANTO")

Escreva o nome da cidade:santo amaro
Nome da cidade comece com SANTO


## Questao 25

### Crie um programa que leia o nome de uma pessoa e diga se ela tem "SILVA" no nome.

In [18]:
nome = input("Digite o nome: ")
nome = nome.upper()
#silva = nome.split()
#print(silva)
if 'SILVA' in silva:
    print("Ela tem 'SILVA' no nome.")
else:
    print("Ela NAO tem 'SILVA' no nome.")

Digite o nome: Santos da SIlva
['SILVA', 'DOS', 'SANTOS']
Ela tem 'SILVA' no nome.


## Questao 26

### Faca um programa que leia uma frase pelo teclado e mostre:

<b> * Quantas vezes parece a letra "A"
<br> * EM que posicao ela aparece a primeira vez.
<br> * EM que posicao ela aparece a ultima vez.

In [36]:
frase = input("Escreva alguma frase: ")
frase = frase.upper()
contador = 0
primeira = -1
ultima = 0
print(frase)
for i in range(len(frase)):
    print(frase[i])
    if frase[i] == 'A':
        contador += 1
        if primeira == -1:
            primeira = i
        if i+1 < len(frase) and 'A' not in frase[(i+1):]:
            print(frase[(i+1):])
            ultima = i

print("Na frase , tem %d vezes a letra 'A' , a primeira vez aparece no lugar %d e, ultima vez aparece\
      no lugar %d"%(contador, (primeira+1), ultima+1))

Escreva alguma frase: a1a2a3
A1A2A3
A
1
A
2
A
3
3
Na frase , tem 3 vezes a letra 'A' , a primeira vez aparece no lugar 1 e, ultima vez aparece      no lugar 5


## Questao 27

### Faca um programa que leia o nome completo de uma pessoa, mostrando em seguida o primeiro e o ultimo nome separadamente.

**Ex: Ana Maria de Souza**
<br> primeiro = **Ana**
<br> ultimo = **Souza**

In [3]:
nome = input("Digite o nome: ")
nome = nome.split()
print("Primeiro nome eh %s e ultimo nome eh %s"%(nome[0], nome[-1]))

Digite o nome: Nome1 nome2 nome3 nome4
Primeiro nome eh Nome1 e ultimo nome eh nome4


In [1]:
lista = [1,2,4,5]
lista[-1]

5

## Questao 28

### Escreva um programa que faca o computador "pensar" em um numero inteiro entre 0 e 5 e peca para usuario tentar descobrir qual foi o numero escolhido pelo computador. O programa devera escrever na tela se o usuario venceu ou perdeu.

In [5]:
import random
numero = random.randint(0,5)
numero_usuario = int(input("Digite numero entre 0 e 5: "))
if numero == numero_usuario:
    print("Venceu")
else:
    print("Perdeu, numero do computador eh - ", numero)

Digite numero entre 0 e 5: 3
Perdeu, numero do computador eh -  1


## Questao 29

### Escreva um programa que leia a velocidade de um carro.
### Se ela ultrapassar 80km/h, mstra uma mensagem dizendo que ele foi multado.
### A multa vai custar RS7,00 por cada km acima do limite.

In [6]:
vel = int(input("Digite a velocidade: "))
multa = (vel - 80)*7
if multa >0 :
    print("Voce ultrapassou a velocidade 80km/h e tem multa de RS%d,00"%multa)
else:
    print("Voce nao ultrapassou")

Digite a velocidade: 126
Voce ultrapassou a velocidade 80km/h e tem multa de RS322,00


## Questao 30

### Crie um programa qie leia um numero intiero e mostre na tela se ele eh PAR ou IMPAR

In [8]:
numero = int(input("Digite um numero: "))
if numero%2 == 0:
    print("O numero %d eh par"%numero)
else:
    print("O numero %d eh impar"%numero)

Digite um numero: 34
O numero 34 eh par


## Questao 31

### Desenvolva um programa que pergunte a distancia de uma viagem em km. Calcule o preco da passagem, cobrando RS0,50 por km para viagen de ate 200km e RS0,45 para viagens mais longas.

In [9]:
distancia = int(input("Digite distancia percorrida: "))
preco = 0
if distancia > 200:
    preco = 100 + (distancia - 200)* 0.45
else: 
    preco = distancia * 0.5
    
print("O preco da passagem eh RS%f."%preco)

Digite distancia percorrida: 342
O preco da passagem eh RS163.900000.


## Questao 32

### Faca um programa que leia um ano qualquer e mostre se ele eh bissexto.

In [12]:
ano = int(input("Digite o ano: "))
if ano %400 == 0:
    print("Este ano eh bissexto.")
elif ano % 4 == 0 and ano % 100 != 0 :
    print("Este ano eh bissexto.")
else:
    print("Este ano NAO eh bissext.")

Digite o ano: 1900
Este ano NAO eh bissext.


## Questao 33

### Faca um programa que leia tres numeros e mostre qual eh o maior e qual eh o menor

In [15]:
n1 = int(input("Digite o primeiro numero: "))
n2 = int(input("Digite o segundo numero: "))
n3 = int(input("Digite o terceiro numero: "))

lista = [n1, n2, n3]
lista.sort()
menor = lista[0]
maior = lista[-1]

print("O maior numero eh %d e o menor eh %d."%(maior,menor))

Digite o primeiro numero: 3
Digite o segundo numero: 4
Digite o terceiro numero: 6
O maior numero eh 6 e o menor eh 3.


## Questao 34

### Escreva um programa que pergunte o salario de um funcionario e calcule o valor do seu aumento. Para salarios superiores  RS1.250,00 , calcule um aumento de 10%. Para os inferiores ou iguais, o aumento eh de 15%.

In [18]:
salario = float(input("Digite o salario: "))

aumento = 0

if salario > 1250:
    aumento = salario*0.1
else:
    aumento = salario*0.15
    
print("Aumento do salario eh %f ."%aumento)

Digite o salario: 2422
Aumento do salario eh 242.200000 .


## Questao 35

### Desenvolva um programa que leia o comprimento de tres retas e diga ao usuario se elas podem ou nao formar um triangulo.

In [19]:
a = float(input("Digite o comprimento da primeira reta: "))
b = float(input("Digite o comprimento da segunda reta: "))
c = float(input("Digite o comprimento da terceira reta: "))

if (a+b>c and a+c>b and b+c>a):
    print("Sim, podemos contruir um triangulo com essas tres retas.")
    
else:
    print("Nos NAO coseguimos construir um triangulo com essas tres retas.")

Digite o comprimento da primeira reta: 3
Digite o comprimento da segunda reta: 4
Digite o comprimento da terceira reta: 6
Sim, podemos contruir um triangulo com essas tres retas.


## Questao 36

### Escreva um programa para aprovar o emprestimo bancario para a compra de uma casa. Pergunte o valor da casa, o salario do comprador e em quantos anos ele vai pagar. A prestacao mensal, nao pode exceder 30% do salario ou entao o emrestimo sera negado.

In [22]:
vcasa = float(input("Qual eh valor da casa? - "))
salario = float(input("Qual eh o valor do salario? - "))
q_anos = int(input("Em quantos anos vai pagar? - "))

pres_mensal = vcasa / (12*q_anos)

if pres_mensal >= 0.3*salario:
    print("EMprestimo negado.")
    
else: 
    print("Emprestimo aprovado.")

Qual eh valor da casa? - 45000
Qual eh o valor do salario? - 5
Em quantos anos vai pagar? - 7
EMprestimo negado.


## Questao 37

### Escreva um programa que leia um numero inteiro qualquer e peca para o usuario escolher qual sera a base de conversao: <br><center> 1.para binario, <br> 2. para octal <br> 3. para hexadecimal

In [30]:
n = int(input("Digite n: "))
base = int(input(' Digite : \n 1 - para binario; \n 2 - para octal; \n 3 - pra hexadecimal '))

if base == 1:
    print(bin(n))
elif base == 2:
    print(oct(n))
elif base == 3:
    print(hex(n))
    
else:
    print("Escreveu base errado.")

Digite n: 15
 Digite : 
 1 - para binario; 
 2 - para octal; 
 3 - pra hexadecimal 3
0xf


## Questao 38

### Escreva um programa que leia dois numeros inteiros e compare-os mostrando na tela uma mensagem: <br><center> * O primeiro valor eh maior <br> * O segundo valor eh maior <br> * Nao existe o valor maior, os dois sao iguais.

In [31]:
primeiro = int(input("Digite o primeiro numero: "))
segundo = int(input("Digite o segundo numero: "))

if primeiro > segundo: 
    print("Primeiro eh maior que segundo")
elif primeiro < segundo :
    print("O segundo eh maior que primeiro")
else:
    print("OS dois sao iguias")

Digite o primeiro numero: 3
Digite o segundo numero: 5
O segundo eh maior que primeiro


## Questao 39

### Faca um programa que leia o ano de nascimento de um jovem e informe, de acordo com sua idade. se ele ainda vai se alistar ao servico militar, se eh a hora de se alistar ou se ja passou do tempo do alistamento. Seu programa tambem devera mostrar tempo que falta ou que passou do prazo.

In [32]:
nascimento = int(input("Digiteo ano do nascimento: "))
idade = 2022 - nascimento

if idade < 18:
    print("Voce ainda vai se alistar ao servico militar, falta %d anos."%(18-idade))
elif idade == 18:
    print(" Hora de alistar ao servico militar.")
else:
    print("Ja passou do tempo do alistamento, passou %d anos do prazo."%(idade-18))

Digiteo ano do nascimento: 1996
Ja passou do tempo do alistamento, passou 8 anos do prazo.


## Questao 40

### Crie um programa que leia duas notas de um aluno e calcule sua media, mostrando uma mansagem no final, de acordo com media atingido: <br><center> * Media abaixo  de 5.0 : REPROVACAO <br> * MEdia entre 5.0 e 6.9: RECUPERACAO <br> * Media 7.0 ou superior: APROVADO

In [33]:
nota1 = float(input("Qual eh primeira nota do alunno? - "))
nota2 = float(input("Qual eh segunda nota do alunno? - "))

media = (nota1 + nota2)/2

if media < 5.0 :
    print("REPROVADO.")
elif 5.0 <= media < 7.0:
    print("RECUPErACAO.")
else: 
    print("AproVADO.")

Qual eh primeira nota do alunno? - 4.5
Qual eh segunda nota do alunno? - 7.9
RECUPErACAO.


## QUestao 41

### A confederacao Nacional de Natacao precisa de um programa que leia o ano de nascimento de um atleta e mostre sua categoria, de acordo com a idade: <br><center> * Ate 9 anos: MIRIM. <br> * Ate 14 anos: INFANTIL. <br> * Ate 19 anos: JUNIOR. <br> * Ate 25 anos: SENIOR. <br> Acima: MASTER

In [34]:
nascimento = int(input("Digite o ano de nascimento: "))
idade = 2022 - nascimento

if idade < 9 :
    print("MIRIM")
elif idade < 14:
    print("INFANtiL")
elif idade < 19:
    print("JUNIOR")
elif idade < 25: 
    print("SENIOR")
else:
    print("MASter")

Digite o ano de nascimento: 1996
MASter


## Questao 42

### Refaca o DESAFIO 35 dos triangulos, acrescentando o recurso de mostrar que tipo de triangulo sera formado: <br> <center> * Equilatero : todos os lados iguais. <br> * Isosceles : Dois ldos iguais. <br> * Escaleno : Todos os lados diferentes

In [35]:
a = float(input("Digite o comprimento da primeira reta: "))
b = float(input("Digite o comprimento da segunda reta: "))
c = float(input("Digite o comprimento da terceira reta: "))

if (a+b>c and a+c>b and b+c>a):
    print("Sim, podemos contruir um triangulo com essas tres retas.")
    
    if a == b and b == c: 
        print("Nossa triangulo eh equilatero.")
    elif a==b or a==c or b==c:
        print("Nosso triangulo eh isosceles.")
    else:
        print("Nosso triangulo eh escaleno.")
else:
    print("Nos NAO coseguimos construir um triangulo com essas tres retas.")

Digite o comprimento da primeira reta: 3
Digite o comprimento da segunda reta: 4
Digite o comprimento da terceira reta: 5
Sim, podemos contruir um triangulo com essas tres retas.
Nosso triangulo eh escaleno.


## Questao 43

### Desenvolvaa uma logica que leia o peso e altura de uma pessoa, calcule seu IMC e mostre seu status, de acordo com a tabela abaixo: <br> <center> * Abaixo de 18.5 : Abaixo do peso. <br> * Entre 18.5 e 25 : Peso Ideal. <br> * 25 ate 30: Sobrepeso. <br> * 30 ate 40: Obesidade. <br> * Acima de 40: Obesidade morbida

In [36]:
peso = float(input("Digite o peso em kg: "))
altura = float(input("Digite a altura em m:  "))

imc = peso / (altura*altura)

if imc < 18.5:
    print("Abaixo do peso.")
elif imc < 25:
    print("Peso Ideal.")
elif imc < 30 : 
    print("Sobrepeso.")
elif imc < 40:
    print("Obesidade.")
else:
    print("Obesidade morbida.")

Digite o peso em kg: 65
Digite a altura em m:  1.83
Peso Ideal.


## Questao 44

### Elabore um prohrama que calcule o valor a ser pago por um produto, considerando o seu preco normal e condicao de pagamento: <br><center> * A vista dinheiro/cheque : 10% de desconto. <br> * A vista no cartao: 5% de desconto. <br> * EM ate 2x no cartao: preco normal. <br> * 3x ou mais no cartao: 20% de juros

In [37]:
preco = float(input("Qual eh o preco normal do produto? - "))
mensagem = "COmo sera o pagamento? Digite: \n 1 - para dinheiro/cheque. \n 2 - A vista no cartao. \n \
3 - em ate 2x no cartao. \n 4 - 3x ou mais no cartao"
pagamento = int(input(mensagem))

valor = 0
if pagamento == 1:
    valor = preco*0.9
elif pagamento == 2:
    valor = preco*0.95
elif pagamento == 3:
    valor = preco
else:
    valor = preco*1.2
    
print("Valor a ser pago eh RS%f ."%valor)

Qual eh o preco normal do produto? - 45
COmo sera o pagamento? Digite: 
 1 - para dinheiro/cheque. 
 2 - A vista no cartao. 
 3 - em ate 2x no cartao. 
 4 - 3x ou mais no cartao4
Valor a ser pago eh RS54.000000 .


## QUestao 45

### Crie um programa que faca o computador jogar Jokenpo com voce.

In [44]:
import random
escolha = ['Pedra', 'Papel', 'Tesoura']
computador = random.randint(0,2)
mensagem = "Qual eh sua escolha? Digite - \n 1 - para Pedra; \n 2 - para Papel; \n 3 - para Tesoura"
usuario = int(input(mensagem))
usuario -= 1

while computador == usuario:
    print('O computador tmb escolheu %s, escolha de novo.'%escolha[usuario])
    usuario = int(input(mensagem))
    usuario -= 1
    computador = random.randint(0,2)
    

if (computador == 0 and usuario == 1) or (computador == 1 and usuario == 2) or (computador == 2 and usuario == 0):
    print("Voce ganhoum pois computador escolheu %s ."%escolha[computador])
    
else:
    print("O coputador ganhou pois 'ELE' escolheu %s ."%escolha[computador])

Qual eh sua escolha? Digite - 
 1 - para Pedra; 
 2 - para Papel; 
 3 - para Tesoura2
O computador tmb escolheu Papel, escolha de novo.
Qual eh sua escolha? Digite - 
 1 - para Pedra; 
 2 - para Papel; 
 3 - para Tesoura2
Voce ganhoum pois computador escolheu Pedra .


## Questao 46 

### Faca um programa que mostre na tela uma contagem regressiva para o estouro de fogod artificio, indo de 10 ate 0, com uma pausa de 1 segundo entre eles.

In [1]:
import time

for i in range(11):
    print(10 - i)
    time.sleep(1)

10
9
8
7
6
5
4
3
2
1
0


## Questao 47

### Crie um programa que mostre na tela todos os numeros pares que estao no intervalo entre 1 e 50.

In [3]:
for i in range(1, 50):
    if i%2 == 0:
        print(i)

2
4
6
8
10
12
14
16
18
20
22
24
26
28
30
32
34
36
38
40
42
44
46
48


## Questao 48

### Faca um programa que calcule a soma entre os numeros impares que sao multiplos de tres e que se encontram no intervalo de 1 ate 500.

In [4]:
n = int(input("Digite 500 ou outro nunero: "))
soma = 0
for i in range(1,n):
    if i%2 == 1 and i%3 == 0:
        print("%d + %d = %d ."%(soma, i, soma+i))
        soma+=i

Digite 500 ou outro nunero: 50
0 + 3 = 3 .
3 + 9 = 12 .
12 + 15 = 27 .
27 + 21 = 48 .
48 + 27 = 75 .
75 + 33 = 108 .
108 + 39 = 147 .
147 + 45 = 192 .


## Questao 49

### Refaca o DESAFIO 9, mostrando a tabuada de um numero que o usuario escolher, so que agora utilizando um laco for.

In [5]:
n = int(input('Digite o numero que ver sua tabuada: '))

print('=================================')
print("*********Tabuada de %d*********"%n)
print('=================================')

for i in range(10):
    print('          %d x %d = %d          '%(n,i+1, n*(i+1)))

Digite o numero que ver sua tabuada: 4
*********Tabuada de 4*********
          4 x 1 = 4          
          4 x 2 = 8          
          4 x 3 = 12          
          4 x 4 = 16          
          4 x 5 = 20          
          4 x 6 = 24          
          4 x 7 = 28          
          4 x 8 = 32          
          4 x 9 = 36          
          4 x 10 = 40          


## Questao 50

### Desenvolva um program que leia seis numeros inteiros e mostre a soma apenas daquelas que forem pares. Se o valor digito for impar, desconsidera-o.

In [6]:
numero = 0
soma = 0
for i in range(6):
    numero = int(input("Digite %d-esimo numero: "%(i+1)))
    if numeros%2 == 0:
        soma += numero
        
print(soma)

Digite 1-esimo numero: 1
Digite 2-esimo numero: 2
Digite 3-esimo numero: 3
Digite 4-esimo numero: 4
Digite 5-esimo numero: 5
Digite 6-esimo numero: 6
12


## Questao 51

### Desenvolva um programa que leia o numero primeiro termo e a razao de uma PA (Progressao Aritmetica) . No final, mostre os 10 primeiros termos dessa progressao.

In [9]:
primeiro = float(input("Primeiro termo da PA: "))
razao = float(input("razao da PS: "))

soma = 0

for i in range(10):
    termo = primeiro + razao*(i)
    print("%d - esimo termo  da PA eh %f. "%(i+1,termo))
    soma += termo
    
print(soma)

Primeiro termo da PA: 1
razao da PS: 1
1 - esimo termo  da PA eh 1.000000. 
2 - esimo termo  da PA eh 2.000000. 
3 - esimo termo  da PA eh 3.000000. 
4 - esimo termo  da PA eh 4.000000. 
5 - esimo termo  da PA eh 5.000000. 
6 - esimo termo  da PA eh 6.000000. 
7 - esimo termo  da PA eh 7.000000. 
8 - esimo termo  da PA eh 8.000000. 
9 - esimo termo  da PA eh 9.000000. 
10 - esimo termo  da PA eh 10.000000. 
55.0


## Questao 52

### Faca um programa que leia um numero inteiro e diga se ele eh ou nao um numero primo.

In [11]:
n = int(input("Digite u mnuero: "))
raiz = int(n**(1/2))
primo = True
i = 2
while not primo and i <= raiz:
    if n%i == 0:
        primo = False
    else:
        i += 1
if primo :
    print(" O numero %d eh primo."%n)
else:
    print(" O numero %d NAO eh primo."%n)

Digite u mnuero: 217
 O numero 217 eh primo.


## Questao 53

### Crie um programa que leia uma frase qualquer e diga se ela eh um palindromo, desconsiderando os espacos.

In [15]:
frase = input("Digite uma frase: ")

polindromo = True
for i in range(len(frase)):
    if frase[i] != frase[len(frase)-1-i]:
        polindromo = False
        
if polindromo :
    print("Nossa frase %s eh polindromo."%frase)
else:
    print("Nossa frase %s NAO eh polindromo."%frase)

Digite uma frase: aba
Nossa frase aba eh polindromo.


## Questao 54

### Crie um programa que leia o ano de nascimento de sete pessoas. No final, mostre quantas pessoas ainda nao atingiram a maioridade e quantas ja sao maiores.

In [16]:
maior = 0
for i in range(1,8):
    ano = int(input("Digite o ano de nascimento da %d- esima pessoa: "%i))
    if ano < 18:
        maior += 1
print("Sao %d pessoas ainda nao atingiram na maioridade."%maior)


Digite o ano de nascimento da 1- esima pessoa: 2
Digite o ano de nascimento da 2- esima pessoa: 3
Digite o ano de nascimento da 3- esima pessoa: 4
Digite o ano de nascimento da 4- esima pessoa: 5
Digite o ano de nascimento da 5- esima pessoa: 15
Digite o ano de nascimento da 6- esima pessoa: 35
Digite o ano de nascimento da 7- esima pessoa: 54
Sao 5 pessoas ainda nao atingiram na maioridade.


## Questao 55

### Faca um programa que leia o peso de cinco pessoas. No final, mostre qual foi a maior e o menor peso lidos.

In [17]:
menor = 10000
maior = -1
for i in range(1,6):
    peso = int(input("DIgite peso da %d-esima pessoa: "%i))
    if peso < menor:
        menor = peso
        
    if peso > maior:
        maior = peso
        
print("O maior peso eh %d e o menor peso eh %d."%(maior,menor))

DIgite peso da 1-esima pessoa: 34
DIgite peso da 2-esima pessoa: 5
DIgite peso da 3-esima pessoa: 35
DIgite peso da 4-esima pessoa: 2
DIgite peso da 5-esima pessoa: 42
O maior peso eh 42 e o menor peso eh 2.


## Questao 56

### Desenvolva um programa que leia o nome, idade e sexo de 4 pessoas. No final do programa, mostre: <br><center> - A media da idade do grupo. <br> - QUal eh o nome do homem mais velho. <br> - Quantas mulheres tem menos de 20 anos.

In [18]:
nome = [0]*4
idade = [0]*4
sexo = [0]*4

soma = 0
velho = 0
mulheres = 0

for i in range(1,5):
    nome[i-1] = input("Digite o nome da %d-esima pessoa: "%i)
    idade[i-1] = int(input("Digite a idade da %d-esima pessao: "%i))
    sexo[i-1] = input("Digite o sexo da %d-esima pessoa (M ou F): "%i)
                      
    soma += idade[i-1]
    if sexo[i-1] == 'M' and idade[i-1]>=idade[velho]:
        velho = i-1
        
    if sexo[i-1] == 'F' and idade[i-1] < 20:
        mulheres += 1
        
print("A media da idade do grupo eh %f."%(soma/4))
print("O nome do homem mais velho eh %s."%(nome[velho]))
print("Tem %d mulheres com idade menor 20."%mulheres)

Digite o nome da 1-esima pessoa: M
Digite a idade da 1-esima pessao: 23
Digite o sexo da 1-esima pessoa (M ou F): M
Digite o nome da 2-esima pessoa: K
Digite a idade da 2-esima pessao: 42
Digite o sexo da 2-esima pessoa (M ou F): F
Digite o nome da 3-esima pessoa: 1
Digite a idade da 3-esima pessao: 1
Digite o sexo da 3-esima pessoa (M ou F): F
Digite o nome da 4-esima pessoa: T
Digite a idade da 4-esima pessao: 23
Digite o sexo da 4-esima pessoa (M ou F): M
A media da idade do grupo eh 22.250000.
O nome do homem mais velho eh T.
Tem 1 mulheres com idade menor 20.


## QUestao 57

### Faca um programa que leia  o sexo de uma pessoa, mas so aceite os valores "M" ou "F". Caso esteja errado, peca a digitacao novamente ate ter um valor correto.

In [20]:
escreveu_certo = False
while not escreveu_certo:
    sexo = input("Digite sexo da pessoa, M ou F: ")
    if sexo == 'M' or sexo == 'F':
        escreveu_certo = True
    else:
        print("Escreveu errado")

Digite sexo da pessoa, M ou F: k
Escreveu errado
Digite sexo da pessoa, M ou F: w
Escreveu errado
Digite sexo da pessoa, M ou F: f
Escreveu errado
Digite sexo da pessoa, M ou F: F


## Questao 58

### Melhore o jogo do DESAFIO 28 onde computador vai "pensar" em um numero entre 0 e 10. So que agora o jogador vai tentar adivinhar ate acertar, mostrando no finl quantos palpites foram necssarios para vencer.

In [22]:
'''
## Questao 28

### Escreva um programa que faca o computador "pensar" em um numero 
inteiro entre 0 e 5 e peca para usuario tentar descobrir qual foi o 
numero escolhido pelo computador. O programa devera escrever na tela 
se o usuario venceu ou perdeu.
'''

import random
numero = random.randint(0,10)
contador = 0
usuario = int(input("Digite o numero entre 0 e 10: "))
while numero != usuario:
    contador += 1
    usuario = int(input("Adivinhou errado, tenta de novo"))
    
print("Voce acertou no %d-esima tentativa."%contador)

Digite o numero entre 0 e 10: 0
Adivinhou errado, tenta de novo1
Adivinhou errado, tenta de novo2
Adivinhou errado, tenta de novo3
Adivinhou errado, tenta de novo4
Adivinhou errado, tenta de novo5
Voce acertou no 5-esima tentativa.


## Questao 59

### Crie um program que leia dois valores e mostre um menu como o ao lado na tela: <br> Seu programa devera realizar a operacao solicitada em cada caso.<br><br>[1]somar<br>[2]multiplicar<br>[3]maior<br>[4]novos numeros<br>[5]sair do programa

In [34]:
n1 = int(input("Digite o primeiro numero: "))
n2 = int(input("Digite o segundo numero: "))
n = [n1, n2]
mult = 1

mensagem = "Digite o numero da operacao: \n \
[1]somar \n [2]multiplicar \n [3]maior \n [4]novos numeros \n [5]sair do programa"

operacao = int(input(mensagem))

while operacao != 5 :
    if operacao == 1:
        print("A soma dos numeros sao iguais a - %d."%sum(n))
    elif operacao == 2:
        for i in n:
            mult *= i
        print("A multiplicacao dos numeros sao iguais a  - %d. "%mult)
        mult = 1
    elif operacao == 3:
        print("O maior dos numeros eh %d."%max(n))
    elif operacao == 4:
        numero = int(input("Digite novo numero: "))
        n.append(numero)
    else:
        print("Nao tem tal opcao, digite de novo.")
    
    operacao = int(input(mensagem))


Digite o primeiro numero: 1
Digite o segundo numero: 2
Digite o numero da operacao: 
 [1]somar 
 [2]multiplicar 
 [3]maior 
 [4]novos numeros 
 [5]sair do programa2
A multiplicacao dos numeros sao iguais a  - 2. 
Digite o numero da operacao: 
 [1]somar 
 [2]multiplicar 
 [3]maior 
 [4]novos numeros 
 [5]sair do programa2
A multiplicacao dos numeros sao iguais a  - 2. 
Digite o numero da operacao: 
 [1]somar 
 [2]multiplicar 
 [3]maior 
 [4]novos numeros 
 [5]sair do programa2
A multiplicacao dos numeros sao iguais a  - 2. 
Digite o numero da operacao: 
 [1]somar 
 [2]multiplicar 
 [3]maior 
 [4]novos numeros 
 [5]sair do programa3
O maior dos numeros eh 2.
Digite o numero da operacao: 
 [1]somar 
 [2]multiplicar 
 [3]maior 
 [4]novos numeros 
 [5]sair do programa4
Digite novo numero: 2
Digite o numero da operacao: 
 [1]somar 
 [2]multiplicar 
 [3]maior 
 [4]novos numeros 
 [5]sair do programa2
A multiplicacao dos numeros sao iguais a  - 4. 
Digite o numero da operacao: 
 [1]somar 
 [2]

## Questao 60

### Faca um programa que leia um numero qualquer e mostre o seu fatorial. 

**Ex:** 
<br> 5! = 5x4x3x2x1 = 120

In [35]:
n = int(input("Digite n: "))
fatorial = 1
for i in range(1,n+1):
    fatorial *= i
    
print("%d! = %d"%(n,fatorial))

Digite n: 5
5! = 120


## Questao 61

### Refaca o DESAFIO 51, lendo o primeiro termo e a razao de uma PA, mostrando os 10 primeiros termos da progressao usando a estrutura *while*.

Questao 51

Desenvolva um programa que leia o numero primeiro termo e a razao de uma PA (Progressao Aritmetica) . No final, mostre os 10 primeiros termos dessa progressao.

In [36]:
primeiro = float(input("Digite o primeiro numero da PA : "))
razao = float(input("Digite a razao da PA : "))
termo = primeiro
contador = 1
while contador <= 10:
    print("%d-esimo termo da PA eh - %f."%(contador,termo))
    termo += razao
    contador += 1

Digite o primeiro numero da PA : 1
Digite a razao da PA : 1
1-esimo termo da PA eh - 1.000000.
2-esimo termo da PA eh - 2.000000.
3-esimo termo da PA eh - 3.000000.
4-esimo termo da PA eh - 4.000000.
5-esimo termo da PA eh - 5.000000.
6-esimo termo da PA eh - 6.000000.
7-esimo termo da PA eh - 7.000000.
8-esimo termo da PA eh - 8.000000.
9-esimo termo da PA eh - 9.000000.
10-esimo termo da PA eh - 10.000000.


## Questao 62

### Melhore o DESAFIO 61, perguntando para o usuario se ele quer mostrar mais alguns termos/ O programa encerra quando ele disser que quer mostrar 0 termos.

In [37]:
primeiro = float(input("Digite o primeiro numero da PA : "))
razao = float(input("Digite a razao da PA : "))
termo = primeiro
contador = 1
while contador <= 10:
    print("%d-esimo termo da PA eh - %f."%(contador,termo))
    termo += razao
    contador += 1
    
mais = int(input("Mais quantos termos quer ver? - "))
Mais = 10
while mais != 0 and contador <= Mais + mais :
    print("%d-esimo termo da PA eh - %f."%(contador,termo))
    termo += razao
    contador += 1
    if contador == Mais + mais +1 : 
        mais = int(input("Mais quantos termos quer ver? - "))
        Mais += mais

Digite o primeiro numero da PA : 1
Digite a razao da PA : 1
1-esimo termo da PA eh - 1.000000.
2-esimo termo da PA eh - 2.000000.
3-esimo termo da PA eh - 3.000000.
4-esimo termo da PA eh - 4.000000.
5-esimo termo da PA eh - 5.000000.
6-esimo termo da PA eh - 6.000000.
7-esimo termo da PA eh - 7.000000.
8-esimo termo da PA eh - 8.000000.
9-esimo termo da PA eh - 9.000000.
10-esimo termo da PA eh - 10.000000.
Mais quantos termos quer ver? - 2
11-esimo termo da PA eh - 11.000000.
12-esimo termo da PA eh - 12.000000.
Mais quantos termos quer ver? - 2
13-esimo termo da PA eh - 13.000000.
14-esimo termo da PA eh - 14.000000.
Mais quantos termos quer ver? - 0


## Questao 63

### Escreva um progrma que leia um numero n inteiro qualquer e mostre na tela primeiros elementos de uma Sequencia Fibonacci.
**Ex:**
0->1->1->2->3->5->8...

In [40]:
n = int(input("Primeiros quantos elementos da sequencia? - "))

def Fibonacci(x):
    if (x == 1 or x == 2) :
        return 1
    else :
        return (Fibonacci(x-1)+Fibonacci(x-2))
    
print("%d-esimo termo da sequencia eh - %d."%(n,Fibonacci(n)))

Primeiros quantos elementos da sequencia? - 8
8-esimo termo da sequencia eh - 21.


## Questao 64

### Crie um programa que leia varios numeros inteiros pelo tecado. O programa so vai parar quando o usuario digitar o valor 999, que e condicao de parada. No final, mostre quantos numeros fora digitados e qual foi a soma entre eles (*desconsiderando o flag*).

In [41]:
numero = int(input("DIgite numero (999 para encerrar)"))
soma = 0
contador = 0

while numero != 999:
    soma += numero
    contador += 1
    numero = int(input("DIgite numero (999 para encerrar)"))
    
print(" Voce digitou %d numeros e soma deles eh %d."%(contador, soma))

DIgite numero (999 para encerrar)1
DIgite numero (999 para encerrar)2
DIgite numero (999 para encerrar)3
DIgite numero (999 para encerrar)4
DIgite numero (999 para encerrar)5
DIgite numero (999 para encerrar)6
DIgite numero (999 para encerrar)999
 Voce digitou 6 numeros e soma deles eh 21.


## Questao 65

### Crie um programa que leia varios numeros inteiros pelo teclado. NO final da execucao, mostre a media entre todos valores e qual foi o maior e o menor valores lidos. O programa deve perguntar ao usuario se ele quer ou nao continuar a digitar valores.

In [46]:
import math
numero = int(input("DIgite numero : "))
soma = 0
contador = 0
menor = math.inf
maior = -menor

continuar = True

while continuar:
    soma += numero
    contador += 1
    if numero < menor:
        menor = numero
    if numero > maior:
        maior = numero
    
    opcao = int(input("Digite 1 - para continuar, 0 - para encerrar."))
    if opcao == 1:
        numero = int(input("DIgite numero : "))
    else:
        continuar = False
        
print("Voce escreveu %d  numeros, media deles eh -> %f. O maior eh -> %d e o menor eh -> %d. "%(contador,soma/contador, maior, menor))

DIgite numero : 1
Digite 1 - para continuar, 0 - para encerrar.1
DIgite numero : 1
Digite 1 - para continuar, 0 - para encerrar.1
DIgite numero : 1
Digite 1 - para continuar, 0 - para encerrar.1
DIgite numero : -9
Digite 1 - para continuar, 0 - para encerrar.1
DIgite numero : 9
Digite 1 - para continuar, 0 - para encerrar.0
Voce escreveu 5 -> numeros, media deles eh -> 0.600000. O maior eh -> 9 e o menor eh -> -9. 


## Questao 66

### Crie um programa que leia varios numeros inteiros pelo teclado. O programa so vai parar quando o usuario digitar o valor 999, que a condicao de parada. No final, mostre quantos numeros foram digitados e qual foi a soma entre eles (*desconsiderando o flag*).

In [47]:
n = int(input("Digite o numero inteiro (999 para encerrar): "))
contador = 0
soma = 0
while n != 999:
    contador += 1
    soma += n
    n = int(input("Digite o numero inteiro (999 para encerrar): "))
print("Digitou %d numeros e soma deles eh %d."%(contador,soma))

Digite o numero inteiro (999 para encerrar): 1
Digite o numero inteiro (999 para encerrar): 2
Digite o numero inteiro (999 para encerrar): 3
Digite o numero inteiro (999 para encerrar): 4
Digite o numero inteiro (999 para encerrar): 999
Digitou 4 numeros e soma deles eh 10.


## Questao 67

### Faca um programa que mostre tabuada de varios numeros, um de cada vez, para cada vlor digitado pelo usuario. O programa sera interrompido quando o numero solicitado for negativo.

In [49]:
def tabuada(n):
    print('=================================')
    print("*********Tabuada de %d*********"%n)
    print('=================================')

    for i in range(10):
        print('          %d x %d = %d          '%(n,i+1, n*(i+1)))
        
n = int(input("Quer ver tabuada de qual nnumero (digite numero negativo para encerrar)? - "))
while n>=0:
    tabuada(n)
    n = int(input("Quer ver tabuada de qual nnumero (digite numero negativo para encerrar)? - "))


Quer ver tabuada de qual nnumero (digite numero negativo para encerrar)? - 3
*********Tabuada de 3*********
          3 x 1 = 3          
          3 x 2 = 6          
          3 x 3 = 9          
          3 x 4 = 12          
          3 x 5 = 15          
          3 x 6 = 18          
          3 x 7 = 21          
          3 x 8 = 24          
          3 x 9 = 27          
          3 x 10 = 30          
Quer ver tabuada de qual nnumero (digite numero negativo para encerrar)? - 0
*********Tabuada de 0*********
          0 x 1 = 0          
          0 x 2 = 0          
          0 x 3 = 0          
          0 x 4 = 0          
          0 x 5 = 0          
          0 x 6 = 0          
          0 x 7 = 0          
          0 x 8 = 0          
          0 x 9 = 0          
          0 x 10 = 0          
Quer ver tabuada de qual nnumero (digite numero negativo para encerrar)? - 6
*********Tabuada de 6*********
          6 x 1 = 6          
          6 x 2 = 12          
      

## Questao 68

### Faca um programa que jogue par ou impar com computador. O jogo so sera interrompido quando o jogador perder, mostrando o total de vitorias consecutivas que ele conquistou no final do jogo.

In [59]:
import random
comp = random.randint(1,5)
valor = int(input("Digite o valor: "))
escolha = int(input("Bota \n 1 - para IMPAR, 2 - para PAR : "))
contador = 0
perdeu = False
while not perdeu:
    if (escolha == 1 and ((comp+valor)%2 == 0)) or (escolha == 2 and ((comp+valor)%2 == 1)):
        print('computador escolheu %d'%comp)
        print("Voce Perdeu, em total, ganhou %d vezes"%contador)
        perdeu = True
        
    else:
        print('Voce ganhou, coputador esclheu %d. '%comp)
        comp = random.randint(1,5)
        valor = int(input("Digite o valor: "))
        escolha = int(input("Bota \n 1 - para IMPAR, 2 - para PAR : "))
        contador+=1

Digite o valor: 1
Bota 
 1 - para IMPAR, 2 - para PAR : 1
computador escolheu 1
Voce Perdeu, em total, ganhou 0 vezes


## Questao 69

### Crie um programa que leia a idade e o sexo de varios pessoas. A cada pessoa cadastrada, o programa devera perguntar se o usuario quer ou nao continuar. No final, mostre: <br><center> A) Quantas pessoas tem mais de 18 anos. <br> B) Quantos homens foram cadastrados. <br> C) Quantas mulheres tem menos de 20 anos.

In [63]:
idade = int(input('Digite a idade: '))
sexo = input("Aperte M - masculino, F - feminino").upper()
continuar = 'S'
maior = 0
homem = 0
mulher_menor = 0

while continuar == 'S': 
    if idade > 18:
        maior += 1
        
    if sexo == "M":
        homem += 1
        
    if sexo == "F" and idade <20:
        mulher_menor += 1
    continuar = input('Quer continuar ainda? [S/N] - ').upper()
    if continuar == 'S':
        idade = int(input('Digite a idade: '))
        sexo = input("Aperte M - masculino, F - feminino").upper()

    
print('A) %d pessoas tem mais de 18 anos.'%maior)
print('B) %d homens foram cadastrados'%homem)
print('C) %d mulheres tem menos de 20 anos'%mulher_menor)

Digite a idade: 12
Aperte M - masculino, F - femininof
Quer continuar ainda? [S/N] - s
Digite a idade: 23
Aperte M - masculino, F - femininom
Quer continuar ainda? [S/N] - s
Digite a idade: 12
Aperte M - masculino, F - femininof
Quer continuar ainda? [S/N] - n
A) 1 pessoas tem mais de 18 anos.
B) 1 homens foram cadastrados
C) 2 mulheres tem menos de 20 anos


## Questao 70

### Crie um programa que leia o nome e o preco de varios produtos. O programa devera perguntar se o usuario  vai continuar. Np final, mostre: <br> <center> A) Qual eh o total gasto na compra. <br> B) Quantos produtos custam mais de RS1000,00 . <br> C) Qual eh o nome do produto mais barato.

In [65]:
produto = input('Digite o nome do produto comprado: ')
preco = float(input('Qual eh o valor do produto? - '))
continuar = True

gasto = 0
caros = 0
barato = preco
prod_barato = produto

while continuar:
    gasto += preco
    
    if preco > 1000:
        caros += 1
    if preco < barato :
        prod_barato = produto
        
    escolha = input("Quer continuar ainda? [S/N] - ").upper()
    if escolha == 'N':
        continuar = False
        
    else:
        produto = input('Digite o nome do produto comprado: ')
        preco = float(input('Qual eh o valor do produto? - '))

        
print('A) Gasto total eh RS %d.'%gasto)
print('B) %d produtos custam mais de RS1000,00'%caros)
print('C) O produto mais barato eh %s'%prod_barato)

Digite o nome do produto comprado: wg
Qual eh o valor do produto? - 2424
Quer continuar ainda? [S/N] - s
Digite o nome do produto comprado: asg
Qual eh o valor do produto? - 242
Quer continuar ainda? [S/N] - s
Digite o nome do produto comprado: weg
Qual eh o valor do produto? - 24242
Quer continuar ainda? [S/N] - n
A) Gasto total eh RS 26908.
B) 2 produtos custam mais de RS1000,00
C) O produto mais barato eh asg


## Questao 71

### Crie um programa que simule o funcionamento de um  caixa eletronico. No inicio, pergunte ao usuario qual sera o valor sacado (*numero inteiro*) e o programa vai informar quantas cedulas de vada valor sera entregues. <br> OBS: Considere que caixa possui cedulas de RS50 , RS20 , RS10 e RS1.

In [66]:
valor = int(input('Qual sera o valor sacado? - '))

cinquenta = valor//50
valor = valor - cinquenta*50
vinte = valor//20
valor = valor - vinte*20
dez = valor//10
valor = valor - dez*10

print('Voce vai receber \n \
      %d unidades de ceddula RS50,00 \n \
      %d unidades de cedula RS20,00 \n \
      %d unidades de cedula RS10,00 \n \
      %d unidades de cedula RS1,00'%(cinquenta, vinte, dez, valor))

Qual sera o valor sacado? - 225
Voce vai receber 
       4 unidades de ceddula RS50,00 
       1 unidades de cedula RS20,00 
       0 unidades de cedula RS10,00 
       5 unidades de cedula RS1,00


## Questao 72

### Crei um programa que tenha uma tupla totalmente preenchida com uma contagem por extenso, de zero ate vinte. <br> Seu programa devera ler um numero pelo tecledo (*entre 0 e 20*) e mostra-lo por extenso.

## Questao 73

### Crie uma tupla preenchida com os 20 primeirps colocados na Tabela do Campeanato Brasileiro de Futebol, na ordem de colocacao. Depois mostre: <br><center> A) Os primeiros. <br> B) Os ultimos 4 colocados. <br> C) Times em ordem alfabetica. <br> D) EM que posicao esta o time da Chapecoensa.

## Questao 74

### Crie um programa que vai gerar cinco numeros aleatorios e colocar em uma tupla. Depois disso, mostre a listagem de numeros gerados e tambem indique o menor e o mau=ior valor que estao na tupla.

In [67]:
import random
tupla = (random.randint(1,10), random.randint(1,10), random.randint(1,10), random.randint(1,10), random.randint(1,10))
print(tupla)
print('O maior eh : ', max(tupla))

(10, 8, 1, 10, 8)
O maior eh :  10


## Questao 75

### Desenvolva um programa que leia quatro valor pelo teclado e guarde-os em uma tupla. No final, mostre: <br><center> A) Quantas vezes apareceu o valor 9. <br> B) EM que posicao foi digitado o primeito valor 3. <br> C) Quais foram os numeros pares.

In [74]:
valor1 = int(input('Primeiro valor: '))
valor2 = int(input('Segundo valor: '))
valor3 = int(input('Terceiro valor: '))
valor4 = int(input('Quarto valor: '))
tupla = (valor1, valor2, valor3, valor4)

contador9 = 0
primeiro3 = -1
pares = []

for i in range(len(tupla)):
    print(tupla[i])
    if tupla[i] == 9:
        contador9 += 1
    if tupla[i] == 3 and primeiro3 == -1:
        primeiro3 = i
    if tupla[i]%2 == 0:
        pares.append(tupla[i])
        
print('A) %d vezes apareceu o valor 9.'%contador9)
print('B) A posicao do primeiro 3 eh %d'%primeiro3)
print('C) Os numeros pares sao: ', pares)

Primeiro valor: 1
Segundo valor: 9
Terceiro valor: 3
Quarto valor: 8
1
9
3
8
A) 1 vezes apareceu o valor 9.
B) A posicao do primeiro 3 eh 2
C) Os numeros pares sao:  [8]


## Questao 76 

### Crie um programa que tenha uma tupla unica com nomes de produtos e seus respectivos precos, na sequencia. No final, mostre um listagem de precos, organizando os dados em forma tabular.

## Questao 77

### Crie um programa que tenha uma tupla com varias palavras (*nao usar acentos*). Depois disso, voce deve mostrar, para cada palavra, quais sao suas vogais.

## Questao 78

### Faca um programa que leia 5 valores numericos e guarde0os em uma lista. No final, mostre qual foi o maior e o menor valor digitado e as suas respectivas posicoes na lista.

## Questao 79

### Crie um programa onde o usuario possa digitar varios valores numericos e cadste-os em uma lista. Caso o numero ja exista la dentro, ele nao sera adicionado. No final, sera exibidos todos os valores unicos digitados, em ordem crescente.

## Questao 80

### Crie um porgrama onde o usuario possa digitar cinco valores numericos e cadastre-os em uma lista, ja na posicao correta de insercao (*sem usar sort()*). No final, mostre a lista ordenada na tela.

## Questao 81

### Crie um programa que vai ler varios numeros e colocar um uma lista. Depois disso, mostre: <br><center> A) Quantos numeros foram digitados. <br> B) A lista de valores, ordenada de forma decrescente. <br> C) Se o valor 5 foi digitado e esta ou nao na lista.

## Questao 82

### Crie  um programa que vai ler varios numeros e colocar em uma lista. Depois disso, crie duas listas extras que vao conter qpenas os valores pares e os valores impares digitados, respectivamente. Ao final, mostre o conteudo das tres listas geradas.

## Questao 83

### Crie um programa onde o usuario digite uma expressao qualquer que use parenteses. Seu aplicativo devera analisar se a expressao passada esta com parenteses abertos e fechdos na ordem correta.

## Questao 84

### Faca um programa que leia nome e peso de varias pessoas, guardando tudo em uma lista. No final, mostre: <br><center> A) Quantas pessoas foram cadastradas. <br> B) Uma listagem com as pessoas mais pesadas. <br> C) Uma listagem com as pessoas mais leves.

## Questao 85

### Crie um programa onde o usuario possa digitar sete valores numericos e cadastre-os em  uma lista unica que mantenha separados os valores pares e impares. No final, mostre os valores pares e impares em ordem crescente.

## Questao 86

### Crie um programa que crie uma matriz de dimensao 3x3 e preencha com valores lidos pelo teclado. No final, mostre a matriz na tela, vom a formatacao correta.

## Questao 87

### Aprimore o desafio anterior, mostrando no final: <br><center> A) A soma de todos os valores pares digitados. <br> B) A soma dos valores de terceira coluna. <br> C) O maior valor da segunda linha.

## Questao 88

### Faca um programa que ajude um jogador da MEGA SENA a criar palpites. O programa vai perguntar quantos jogos serao gerados e vai sortear 6 numeros entre 1 e 60 para cada jogo, cadastrando tudo em uma lista composta.

## Questao 89

### Crie um programa que leia nome e duas notas de varios alunos e guarde tudo em uma lista composta. No final, mostre um boletim contendo a media de cada um e permita que o usuario possa mostrar as notas de cada aluno individualmetne.

## Questao 90

### Faca um programa que leia nome e media de um aluno, guardando tambem a situacao em um dicionario. No final, nostre o conteudo da estrutura na tela.

## Questao 91

### Crie um programa onde 4 jogadpres joguem um dado e tenham resultados aleatorios. Guarde esses resultados em um dicionairo. No final, coloque esse dicionario em ordem, sabendo que o vencedor tirou o maior numero do dado.

## Questao 92

### Crie um programa que leia nome, ano de nascimento e carteira de trablaho e cadastre-os (*com idade*) em um dicionario se por acaso a CTPS for diferente de ZERO, o dicionario recebera tambem o ano de contratacao e o salario. Calcule e acrescente, alem da idade, com quantos anos a pessoa vai se aposentar.

## Questao 93

### Crie um programa que gerencie o aproveitaento de um jogador de futebol. O programa vai ler nome do jogador e quantas partidas ele jogou. Depois vai ler a quantidade de gols feitos em cada partida. No final,  tudo isso sera guardado em um dicionario, incluindo o total de gols feitos durante o campeonato.

## Questao 94

### Crie um programa que leia nome, sexo e idade de varias pessoas, guardando os dados de cada pessoa em um divionario e todos os dicionarios em uma lista. No final, mostre: <br> <center> A) Quantas pessoas cadastradas. <br> B) A media da idade. <br> C) Uma lista com mulheres. <br> C) Uma lista com idade acima da media.

## Questao 95

### Aprimore o DESAFIO 93 para que ela funcione com varios jogadores, incluindo um sistem de visualizacao de detalhes do aproveitamento de cada jogador.

## QUestao 96

### Faca um programa que tenha uma funcao chamada *AREA()*, que receba as dimensoes de um terreno retangular (Largura e comprimento) e mostre a area do terreno.

## Questap 97

### Faca um programa que tenha uma funcao chamada *escreva()*, que receba um texto qualquer como parametro e mostre uma mensagem vcom tamanha adoptavel.

**Ex:**
<br> *escreva*('Ola, Mundo!')
<br>**Saida:**
<br>--------------
<br>  Ola, Mundo!
<br>--------------

## Questao 98

### Faca um programa que tenha uma funcao chamada *contador()*, que receba tres parametros: iniciom fim e passso. Seu programa tem que realizar tres contagens atraves da funcao criada:<br><center> A) De 1 ate 10, de 1 em 1. <br> B) e 10 ate 9, de 2 em 2. <br> C) Uma contagem personalizada. 

## Questao 99

### Faca um programa que tenha uma funcao chamada *maior()*, que receba varios parametros com valores inteiros. Sei programa tem que analisar todos os valores e dizer qual deles eh o maior.

## Questao 100

### Faca um programa que tenha uma lista chamada numeros e duas funcoes chamadas *sorteia()* e *somaPar()*. A primeira funcao vai sortear 5 numeros e vai coloca-los dentro da lista e segunda funcao vai mostrar a soma entre todos os valores PARES sorteads pela funcao anterior.

## Questao 101

### Crie um programa que tenha uma funcai chamada *voto()* que vai receber como parametro o ano de nascimento da uma pessoa, retornando um valor literal indicando se uma pessia tem voto NEGADOm OPSIONAL ou OBRIGATORIO nas eleicoes.

## Questao 102

### Crie um programa que tenha funcao *fatorial()* que receba dois parametros: o primeiro que indiqueo numero a calcular e outro chamado show, que sera um valor logico(opcional) indicando se sera mostrdo ou nao na tela o processo de calculo do fatorial.

## Questao 103

### Faca um programa que tenha uma funcao chamada *ficha()*, que receba dois parametros opcionais: o nome de um jogdor e quantos gols ele marcou. O programa devera ser capaz de mostrar a ficha dp jogador, mesmo que algum dado nao tenha sido informado corretaente.

## Questao 104 

### Crie um programa que tenha a funcao *leiaInt()*, que vai funcionar de forma semelhante a funcao *input()* do Python, ao que fazendo a validacao para aceitar apenas um valor numerico.

**Ex:**
<br> n = leiaInt("Digite um n")

## Questao 105

### Faca um programa que tenha uma funcao *notas()* que pode receber varias notas de alunos e vai retornar um dicionario com as seguintes informacoes:<br> <center> - Quantidade de notas. <br> - A amior nota. <br> - A menor nota. <br> - A media da turma. <br> - A situacao (opcional) Adicione tambem as docstrings.

## Questao 106

### Faca um mini-sistema que utilize o Interactive Help do Python. O usuario vai digitar o comando e o manual vai aparecer. Quando o usuario digitar a palavra "FIM", o programa se encerrara.

**OBS: use cores**

## Questao 107

### Cria um modulo chamado *moeda.py* que tenha as funcoe incorporadas *aumentar(), diminuir(), dobro() e metade()*. Faca tambem um programa que importe esse modulo e use algumas dessas funcoes.

## Questao 108

### Adapte o codiggo do DESAFIO 107, criando uma funcao adiciona; chamada *moeda()* que consiga mostrar os valores como um valor monetario formatado.

## Questao 109

### Modifique as funcoes foram criadas no DESAFIO 108 para que elas aceitem um parametro a mais, informando se o valor retornafo por elas vai ser ou nao formatado pela funcao *moeda()*, desenvolvida no DESAFIO 108.

## Questao 110

### Adicione ao modulo *moeda.py* criado nos desafios anteriores, uma funcao chamada *resumo()*, que mostre na tela  algumas informacoes geradas pelas funoes quue ja temos no modulo criado ate aqui.

## Questao 111

### Crie um pacote chamado *utilidadeCev* que tenha dois modulos internos chamados moeda e dado. <br> Tranfira todas as funcoes utilizadas nos DESAFIOS 107, 108 e 109 para o primeiro pacote e mantenha tudo funcionando.

## Questao 112

### Dentro do pacote *utilidadesCev* que criamos no DESAFIO 111, temos um modulo chamafo dado. Crie uma funcao chamada *leiaDinheiro()* que seja capaz de funcionar como a funcao *input()*, ms com uma validacao  de dados para aceitar apenas que sejam monetarios.

## Questao 113

### Reescreva a funcao *leiaInt()* que fizemos no DESAFIO 104, incluindo agora a possibilidade de digitacao de um numero de tipo invalido. AProveite e cria tambem uma funcao *leiaFLoat()* com a mesma funcionalidade.

## Questao 114

### Crie um codigo em Python que teste se o site Pudim esta acessivel pelo computador usado.

## Questao 115

### Crie um pequeno sistema modularizado que permita cadastrar pessoas pelo seu nome e idade em um arquivo de texto simples. <br> O sistema so vai ter 2 opcoes: cadastrar uma nova pessoa e listar todas as pessoas cadastradas.